components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp

```diff
@@ -878,5 +880,87 @@ namespace scream {
             });
         }
 
+        void compute_aerocom_cloudtop(
+            int ncol, int nlay, const real2d &tmid, const real2d &pmid,
+            const real2d &p_del, const real2d &z_del, const real2d &qc,
+            const real2d &qi, const real2d &cldfrac_tot, const real2d &nc,
+            real1d &T_mid_at_cldtop, real1d &p_mid_at_cldtop,
+            real1d &cldfrac_ice_at_cldtop, real1d &cldfrac_liq_at_cldtop,
+            real1d &cldfrac_tot_at_cldtop, real1d &nc_at_cldtop) {
+            // Set outputs to zero
+            memset(T_mid_at_cldtop, 0);
+            memset(p_mid_at_cldtop, 0);
+            memset(cldfrac_ice_at_cldtop, 0);
+            memset(cldfrac_liq_at_cldtop, 0);
+            memset(cldfrac_tot_at_cldtop, 0);
+            memset(nc_at_cldtop, 0);
+            // initialize the 1D "clear fraction" as 1 (totally clear)
+            auto aerocom_clr = real1d("aerocom_clr", ncol);
+            memset(aerocom_clr, 1);
+            // initialize helper variable as 0
+            auto aerocom_tmp = real1d("aerocom_tmp", ncol);
+            memset(aerocom_tmp, 0);
+            // initialize the 2D "cloudy fraction" as 0 (totally clear)
+            auto aerocom_cld = real2d("aerocom_cld", ncol, nlay);
+            memset(aerocom_cld, 0);
+            // get gravity constant from constants
+            using physconst = scream::physics::Constants<Real>;
+            // Loop over all columns in parallel
+            yakl::fortran::parallel_for(
+                SimpleBounds<1>(ncol), YAKL_LAMBDA(int icol) {
+                  // Loops over all layers in serial (due to accumulative
+                  // product), starting at 2 (second highest) layer because the
+                  // highest one has no clouds
+                  for(int ilay = 2; ilay <= nlay; ++ilay) {
+                    // We only do the calculation if certain conditions are met
+                    if((qc(icol, ilay) + qi(icol, ilay)) >
+                           0 &&                              // BAD_CONSTANT!
+                       (cldfrac_tot(icol, ilay) > 0.001)) {  // BAD_CONSTANT!
+                      aerocom_cld(icol, ilay) = cldfrac_tot(icol, ilay);
+                      aerocom_tmp(icol) =
+                          aerocom_clr(icol) *
+                          (1 - ekat::impl::max(aerocom_cld(icol, ilay - 1),
+                                               aerocom_cld(icol, ilay))) /
+                          (1 - ekat::impl::min(aerocom_cld(icol, ilay - 1),
+                                               1 - 0.001));  // BAD_CONSTANT!
+                      // T_mid_at_cldtop
+                      T_mid_at_cldtop(icol) =
```

Comment:
Sorry for being dense, but I'm having a hard time understanding these calculations and what they mean physically. Can you add some more description here, at least so I can follow along?